File: Исходники/Глава 3. Часть 1/3. Сортировка вставками/Python/insert sort.py
def insert_sort(array: list, data_length: int):

	for number_index in range(0, data_length):
	
		current_index = number_index
	
		while current_index > 0 and array[current_index] < array[current_index - 1]:
			tmp = array[current_index - 1]
			array[current_index - 1] = array[current_index]
			array[current_index] = tmp
			current_index -= 1

	return array

File: Исходники/Глава 3. Часть 1/3. Сортировка вставками/C++/Продуктовый/insert sort.cpp
template<typename T>
bool comp(T firstData, T secondData)
{
	return firstData < secondData;
}

template<typename T>
T InsertionSort(T *arr, int array_size, bool comp)
{
	int tmp;
	for (int numberIndex = 0; i < array_size; i++)
	{
		int currentIndex = numberIndex;
		
		while (not comp(currentIndex, 0))
		{
			if (comp(arr[currentIndex], arr[currentIndex - 1]))
			{
				tmp = arr[currentIndex - 1];
				arr[currentIndex - 1] = arr[currentIndex];
				arr[currentIndex] = tmp;
				currentIndex--;
			}
		}
	}
	
	return arr;
}